proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,241 @@
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encode(transactionSchema: LiskJSONSchema, trs: Transaction) -> bytes:
+    trs.params = encode(paramsSchema,trs.params)
+    return encode(transactionSchema,trs)
+```
+
+### Deserialization
+
+Consider a binary message `trsMsg`, corresponding to a serialized transaction. The deserialization procedure is as follows:
+
+```python
+def decode(transactionSchema: LiskJSONSchema, trsMsg: bytes) -> Transaction: 
+    trsData = decode(transactionSchema,trsMsg)
+    trsData.params = decode(paramsSchema,trsData.params)
+    return trsData
+```
+
+### Transaction signature calculation
+Consider a data structure `unsignedTrsData` representing a valid `Transaction` object in which the signatures array is initialized to the default value (an empty array). The following function calculates a signature of the object on a certain chain with secret key `sk`. 
+
+```python
+def signTransaction(sk: PrivateKeyEd25519, unsignedTrsData: Transaction) -> SignatureEd25519:
+    serializedTrs = encode(transactionSchema, unsignedTrsData)
+    return signMessage(sk, MESSAGE_TAG_TRANSACTION, networkIdentifier, serializedTrs)
+```
+
+Here `networkIdentifier` is the correct [network identifier](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#network-identifiers) for the chain and the function `signMessage` is defined in [LIP 0037](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#signing-and-verifying-with-ed25519).
+
+
+## Backwards Compatibility
+
+This LIP results in a hard fork as nodes following the proposed protocol will reject transactions following the previous schema, and nodes following the previous protocol will reject transactions following the proposed schema.
+
+
+## Reference Implementation
+
+TBD
+
+
+## Appendix
+
+In this section, we present a serialization example for a transfer transaction. To calculate the signature, we use the network identifier: `networkID = 9ee11e9df416b18bf69dbd1a920442e08c6ca319e69926bc843a561782ca17ee` and the tag: `tag = "LSK_TX_".encode()`.
+
+#### **Transaction object to serialize:**
+
+```java
+myTrs = {
+  "moduleID": '00000002',
+  "commandID": '0000',
+  "nonce": 5n,
+  "fee": 1216299416n,
+  "senderPublicKey": '6689d38d0d89e072b5339d24b4bff1bd6ef99eb26d8e02697819aecc8851fd55',
+  "params": {
+    "amount": 123986407700n,
+    "recipientID": '2ca4b4e9924547c48c04300b320be84e8cd81e4a',
+    "data": 'Odi et amo. Quare id faciam, fortasse requiris.'
+  },
+  "signatures": [
+    '9953f164f9664e05526c1e3a10c4631715cdcb9fd4f376bf7db5334ded3bbc8470bce023d67c7aca16cf3389ea01f3e3c011820c317f1f5a63f98bb6d6b34b07',
+    'a95fc611f7207ddaaaf7929f8f19b7c1cb2473ead20e9be99b8c0abc148b4ea35713ed296acbd6612f124698e96d57e6fde0eddbb998b86203d04ff3c3976700'
+  ]
+}
+```
```

Comment:
The example above needs to be updated (moduleID and commandID).

Reply:
I am aware of that. I will update if I find time, otherwise remove the appendix for now and merge it later.